src/main/java/com/aws/greengrass/shadowmanager/model/configuration/ShadowSyncConfiguration.java

```diff
@@ -225,37 +226,37 @@ private static void processThingShadowSyncConfiguration(Set<ThingShadowSyncConfi
         Validator.validateThingName(thingName);
 
         ThingShadowSyncConfiguration syncConfiguration;
-        boolean syncClassicTopic = true;
-        for (Map.Entry<String, Object> configObjectEntry : thingConfig.entrySet()) {
-            switch (configObjectEntry.getKey()) {
-                case CONFIGURATION_CLASSIC_SHADOW_TOPIC:
-                    syncClassicTopic = Coerce.toBoolean(configObjectEntry.getValue());
-                    break;
-                case CONFIGURATION_NAMED_SHADOWS_TOPIC:
-                    if (configObjectEntry.getValue() instanceof List) {
-                        List<String> namedShadows = Coerce.toStringList(configObjectEntry.getValue());
-                        for (String namedShadow : namedShadows) {
-                            Validator.validateShadowName(namedShadow);
-                            syncConfiguration = ThingShadowSyncConfiguration.builder()
-                                    .thingName(thingName)
-                                    .shadowName(namedShadow)
-                                    .build();
-                            syncConfigurationSet.add(syncConfiguration);
-                        }
-                    } else {
-                        throw new InvalidConfigurationException(String.format(UNEXPECTED_TYPE_FORMAT,
-                                configObjectEntry.getKey(), configObjectEntry.getClass().getTypeName()));
-                    }
-                    break;
-                default:
-                    // Do nothing here since we want to be lenient with unknown fields.
-                    break;
+        boolean addedOnInteraction = Coerce.toBoolean(
+                thingConfig.getOrDefault(CONFIGURATION_ADDED_ON_INTERACTION_TOPIC, false)
+        );
+
+        Object namedShadowsObj = thingConfig.getOrDefault(CONFIGURATION_NAMED_SHADOWS_TOPIC, null);
```

Comment:
Instead of defaulting to null, default to an empty list. That way you can remove the nested if's which increase the code complexity.

Reply:
This part does not exist anymore, thank you for the tip! 👍